<launch>
  <!-- Arguments -->
  <arg name="tb3_name"  default="tb3_0"/>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <rosparam command="load" file="$(find multi_navigation)/config/params.yaml"/>

  <remap from="/$(arg tb3_name)/map" to="/map"/>

  <group ns = "$(arg tb3_name)">
    <param name="tb3_name" value="$(arg tb3_name)" />

    <include file="$(find turtlebot3_bringup)/launch/includes/description.launch.xml">
      <arg name="model" value="$(arg model)" />
    </include>

    <!-- launch state publisher -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg tb3_name)" />
    </node>

    <!-- AMCL -->
    <include file="$(find multi_navigation)/launch/amcl_with_map.launch">
        <arg name="tf_prefix" default="$(arg tb3_name)" />
    </include>

    <!-- detect robot local positions -->
    <node pkg="multi_navigation" name="detect_robots" type="detect_robots.py" output="screen" respawn="true">
      <param name="tb3_name" type = "str" value="$(arg tb3_name)"/>
    </node>

    <!-- avoid -->
    <node pkg="multi_navigation" name="avoid_collision" type="avoid_collision.py">
    </node>

    <!-- vel integrator -->
    <node pkg="multi_navigation" name="vel_publisher" type="vel_publisher.py"></node>

  </group>
</launch>
